recipes/libsodium/all/conanfile.py

```diff
@@ -52,164 +79,94 @@ def _vs_configuration(self):
-        if self.settings.os.version == "7.0" and str(self.settings.arch) in neutrino_archs:
-            host_arch = "%s-nto-qnx7.0.0" % neutrino_archs[str(self.settings.arch)]
-            if self.settings.arch == "armv7":
-                host_arch += "eabi"
-        else:
-            raise ConanInvalidConfiguration("Unsupported arch or Neutrino version for libsodium: {} {}".format(self.settings.os, self.settings.arch))
-        self._build_autotools_impl(configure_args, host_arch)
+        msbuild.build(sln_path, upgrade_project=False, platforms={"x86": "Win32"}, build_type=self._msvc_configuration)
 
-    def _build_autotools(self):
-        absolute_install_dir = os.path.abspath(os.path.join(".", "install"))
-        absolute_install_dir = absolute_install_dir.replace("\\", "/")
-        configure_args = self._get_configure_args(absolute_install_dir)
+    @property
+    def _android_abi_str(self):
+        return "androideabi" if str(self.settings.arch) in ["armv6", "armv7"] else "android"
 
-        if self.settings.os in ["Linux", "FreeBSD"]:
-            self._build_autotools_linux(configure_args)
-        elif self.settings.os == "Emscripten":
-            self._build_autotools_emscripten(configure_args)
-        elif self.settings.os == "Android":
-            self._build_autotools_android(configure_args)
-        elif tools.is_apple_os(self.settings.os):
-            self._build_autotools_darwin(configure_args)
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        yes_no = lambda v: "yes" if v else "no"
+        if self.settings.os in ("FreeBSD", "Linux"):
+            host = None
         elif self._is_mingw:
-            self._build_autotools_mingw(configure_args)
+            host = "{}-w64-mingw32".format("i686" if self.settings.arch == "x86" else self.settings.arch)
+        elif tools.is_apple_os(self.settings.os):
+            host = "{}-apple-{}".format(self.settings.arch, "ios" if self.settings.os == "iOS" else "darwin")
+        elif self.settings.os == "Android":
+            host = "%s-linux-%s" % (tools.to_android_abi(self.settings.arch), self._android_abi_str)
         elif self.settings.os == "Neutrino":
-            self._build_autotools_neutrino(configure_args)
+            neutrino_archs = {"x86_64":"x86_64-pc", "x86":"i586-pc", "armv7":"arm-unknown", "armv8": "aarch64-unknown"}
+            if self.settings.os.version == "7.0" and str(self.settings.arch) in neutrino_archs:
+                host = "{}-nto-qnx7.0.0".format(neutrino_archs[str(self.settings.arch)])
+                if self.settings.arch == "armv7":
+                    host += "eabi"
+            else:
+                raise ConanInvalidConfiguration("Unsupported arch or Neutrino version for libsodium: {} {}".format(self.settings.os, self.settings.arch))
```

Comment:
all these ConanInvalidConfiguration in build() should be removed, I've a PR fot that, but forgot to submit it.

Reply:
I would be completely fine with you submitting your pr modernizing libsodium that would supersede this pr.

Reply:
Let's merge this one, and I'll submit another one to avoid too many modifications in one PR.